### PR TITLE
chore: release script make create_pull all kwargs, better file adding for changelog

### PR DESCRIPTION
**Pull request by ZStriker19**
When I used release script for 2.8 release:
errored out on this step:
```
Traceback (most recent call last):
  File "/Users/zach.groves/projects/dd-trace-py/scripts/release.py", line 489, in <module>
    create_changelog_pull_request(dd_repo, name, rn)
  File "/Users/zach.groves/projects/dd-trace-py/scripts/release.py", line 276, in create_changelog_pull_request
    commit_and_push(dd_repo, f"release.script/changelog-update-{name}", name)
  File "/Users/zach.groves/projects/dd-trace-py/scripts/release.py", line 269, in commit_and_push
    dd_repo.create_pull(DEFAULT_BRANCH, branch_name, title=f"chore: {pr_body}", body=f"- [x] {pr_body}", draft=False)
  File "/Users/zach.groves/projects/dd-trace-py/scripts/venv/lib/python3.9/site-packages/github/Repository.py", line 1399, in create_pull
    return self.__create_pull_1(*args, **kwds)
TypeError: __create_pull_1() got multiple values for argument 'title' 
```

I'm pretty sure this was due to a method signature change: https://github.com/PyGithub/PyGithub/releases/tag/v2.1.0.post0

It also added other uncommitted files like .DS_Store.  So I instead have it just add the changelog file, since that's the only one that should be different.

## Checklist

- [x] Change(s) are motivated and described in the PR description
- [x] Testing strategy is described if automated tests are not included in the PR
- [x] Risks are described (performance impact, potential for breakage, maintainability)
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed or label `changelog/no-changelog` is set
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/))
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))
- [x] If this PR changes the public interface, I've notified `@DataDog/apm-tees`.

## Reviewer Checklist

- [x] Title is accurate
- [x] All changes are related to the pull request's stated goal
- [x] Description motivates each change
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- [x] Testing strategy adequately addresses listed risks
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] Release note makes sense to a user of the library
- [x] Author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: scripts/release.py [hunk 1]
```diff
@@ -261,12 +261,23 @@ def add_release_to_changelog(name: str, release_notes: str):
 
 def commit_and_push(dd_repo, branch_name: str, release_name: str):
     subprocess.check_output(f"git checkout -b {branch_name}", shell=True, cwd=os.pardir)
-    subprocess.check_output("git add -A", shell=True, cwd=os.pardir)
+    try:
+        subprocess.check_output("git add CHANGELOG.md", shell=True, cwd=os.pardir)
```
Comment (brettlangdon) [inline scripts/release.py:265]: `git add $(git rev-parse --show-toplevel)/CHANGELOG.MD)`

I was just curious if you could get the root directory of the repo regardless of which sub-directory you might be in.

Comment (brettlangdon) [reply scripts/release.py:265]: I am not sure `$(command)` works in Python subprocess though....

**Review by brettlangdon — APPROVED** [review]
minor suggestion, but otherwise lgtm

File: scripts/release.py (hunk 1 above)
Comment (emmettbutler) [reply scripts/release.py:265]: Yeah I think for this to work you'd need to create multiple subprocesses and pipe them together. I like the flexibility of that approach, though. Possible future improvement.

Comment (emmettbutler) [reply scripts/release.py:265]: @ZStriker19 maybe line 36 in the docstring of this file should change if it can be run from cwds other than `scripts`.

**Review by emmettbutler — APPROVED** [review]

**Pull request merged**